docs/topics/db/queries.txt

```diff
@@ -825,6 +825,93 @@ being evaluated and therefore populate the cache::
+iteration time. To fix this, you can just swap to ``async for``::
+
+    async for entry in Authors.objects.filter(name__startswith="A"):
+        ...
+
+There's not much more to it than that - just be aware that you also can't do
+other things that might iterate over the QuerySet, such as wrapping ``list()``
+around it to force its evaluation (we ship a ``django.utils.asyncio.alist``
+function that does this asynchronously, if you want it).
+
+Because QuerySet methods like ``filter`` and ``exclude`` do not actually run
+the query - they just set up the QuerySet to run when it's iterated over - you
+can use those freely in asynchronous code. For a guide to which methods can
+keep being used like this, and which have asynchronous versions, read the next
+section.
+
+QuerySet and manager methods
+----------------------------
+
+Some methods on managers and querysets - like ``get`` and ``first`` - force
+execution of the queryset and are blocking. Some, like ``filter`` and
+``exclude``, don't force execution and so are safe to run from asynchronous
+code. But how are you supposed to tell the difference?
+
+While you could just poke around and see if there is an ``a``-prefixed version
+of the method (for example, we have ``aget`` but not ``afilter``), there is
+a more logical way - look up what kind of method it is in our
+:doc:`QuerySet reference </ref/models/querysets>` documentation.
+
+In there, you'll find the methods on QuerySets grouped into two sections:
+
+* *Methods that return new QuerySets*: These are the non-blocking ones, and
+  don't have asynchronous versions. You're free to use these in any situation,
+  though read the notes on ``defer()`` and ``only()`` before you use them.
+
+* *Methods that do not return QuerySets*: These are the blocking ones, and
+  have asynchronous versions - the asynchronous name for each is noted in its
+  documentation, though our standard pattern is to add an ``a`` prefix.
+
+Using this distinction, you can work out when you need to use asynchronous
+versions, and when you don't. For example, here's a valid asynchronous query::
+
+    user = await User.objects.filter(username=my_input).afirst()
+
+``filter`` returns a QuerySet, and so it's fine to keep chaining it inside an
+asynchronous environment, whereas ``first`` evaluates and returns a model
+instance - thus, we change to ``afirst``, and use ``await`` at the front of
+the whole expression in order to call it in an asynchronous-friendly way.
+
+.. note::
+
+    If you forget to put the ``await`` part in, you may see errors like
+    "coroutine object has no attribute x" or ``<coroutine ...>`` strings in
+    place of your model instances. If you ever see these, you are missing an
+    ``await`` somewhere to turn that coroutine into a real value.
+
+Transactions
+------------
+
+Transactions are not currently supported with async queries and updates;
+you will find that trying to use one raises ``SynchronousOnlyOperation``.
```

Comment:
why this is not supported? or looking to handle that later time?

Reply:
I find it best to attack large refactors in small, reasonable chunks. Transaction support can be done separately, so we should, to keep this PR smaller and make things easier to land.